### PR TITLE
OFI: Disable unused caps

### PR DESCRIPTION
**Pull request by jdinan**
Signed-off-by: James Dinan james.dinan@intel.com

### CODE REVIEW

**Comment by jdinan** [comment]
@kseager, @sayantansur - please review

File: src/transport_ofi.c [hunk 1]
```diff
@@ -297,7 +297,15 @@ static inline int allocate_endpoints(struct fabric_info *info)
      * support shmem_wait that blocks on an update arriving from the network */
 #ifndef ENABLE_HARD_POLLING
     info->p_info->caps |= FI_REMOTE_WRITE | FI_REMOTE_READ;
+#else
+    info->p_info->caps &= ~FI_REMOTE_WRITE;
+    info->p_info->caps &= ~FI_REMOTE_READ;
```
Comment (kseager) [inline src/transport_ofi.c]: if you want a send-only you should instead set FI_WRITE and FI_READ

File: src/transport_ofi.c [hunk 2]
```diff
@@ -297,7 +297,15 @@ static inline int allocate_endpoints(struct fabric_info *info)
...
 #endif
+
+    /* Remove unused capabilities */
+    info->p_info->caps &= ~FI_MSG;
+    info->p_info->caps &= ~FI_TAGGED;
```
Comment (kseager) [inline src/transport_ofi.c]: this is unnecessary, we don't need to clear these bits if we never set them in the first place

Comment (sayantansur) [inline src/transport_ofi.c]: OTOH, some providers might just provide you with more functionality than you asked for. No harm in turning them off again.

Comment (kseager) [inline src/transport_ofi.c]: but then whats the point of turning them off? either its a provider bug if having them on the caps changes functionality from what you asked the provider for or there is nothing to gain from turning them off; that's just my two cents

Comment (sayantansur) [inline src/transport_ofi.c]: Yea, ignore this, and probably better to delete this part of the patch.

File: src/transport_ofi.c (hunk 1 above)
Comment (sayantansur) [inline src/transport_ofi.c]: Discussed with Kayla here. Adding permissions to read/write if not hard polling seems wrong to me. You need the permissions to read/write anyways (and already covered under FI_RMA), and hard polling only controls whether you look at the target counter or not e.g. FI_RMA_EVENT.

Comment (kseager) [inline src/transport_ofi.c]: yeah guys what did you do in commit: 082c6747e89626bb6facc719e63db040b82388e7 when I was out of town? ;) just kidding. Agreed we should just be keeping the caps=FI_RMA for the ep creation regardless

Comment (sayantansur) [inline src/transport_ofi.c]: Yea, it looks like I messed up on that one. Line 295-299 in 082c674 should be deleted.

Comment (jdinan) [inline src/transport_ofi.c]: *facepalm*

**Comment by sayantansur** [comment]
This one looks good to me 👍

**Comment by jdinan** [comment]
@kseager Could you please review the updated change set and merge if it look good?

**Pull request merged**